src/axom/slic/interface/slic.cpp

```diff
@@ -24,74 +25,69 @@ void initialize() { Logger::initialize(); }
 bool isInitialized() { return (Logger::getActiveLogger() != nullptr); }
 
 //------------------------------------------------------------------------------
-void createLogger(const std::string& name, char imask)
+void ensureInitialized()
 {
   if(!isInitialized())
   {
-    std::cerr << "[ERROR]: slic::initialize() must be called "
-              << "before making any other calls to SLIC.";
-    return;
+    initialize();
+    disableAbortOnError();
+    disableAbortOnWarning();
+    setLoggingMsgLevel(slic::message::Debug);
+    std::string format = std::string("<TIMESTAMP>\n") +
+      std::string("[<LEVEL>]: <MESSAGE> \n") + std::string("FILE=<FILE>\n") +
+      std::string("LINE=<LINE>\n\n");
+    addStreamToAllMsgLevels(new slic::GenericOutputStream(&std::cout, format));
+
+    logMessage(
+      message::Warning,
+      "slic::initialize() must be called before any other calls to SLIC\n."
```

Comment:
Is calling `slic::initialize()` enough?  What happens if they call it then dont' create a Logger?

Reply:
`slic::initialize()` calls `static Logger::initialize`, which creates and activates Logger "root."  So I think it is enough.